rust/private/rustc.bzl

```diff
@@ -513,6 +528,43 @@ def rustc_compile_action(
         ),
     ]
 
+def establish_cc_info(ctx, crate_info, toolchain, cc_toolchain, feature_configuration):
+    """ If the produced crate is suitable yield a CcInfo to allow for interop with cc rules """
+
+    if crate_info.type not in ("staticlib", "cdylib"):
+        return []
+
+    if crate_info.type == "staticlib":
+        library_to_link = cc_common.create_library_to_link(
+            actions = ctx.actions,
+            feature_configuration = feature_configuration,
+            cc_toolchain = cc_toolchain,
+            static_library = crate_info.output,
+        )
+    elif crate_info.type == "cdylib":
+        library_to_link = cc_common.create_library_to_link(
+            actions = ctx.actions,
+            feature_configuration = feature_configuration,
+            cc_toolchain = cc_toolchain,
+            dynamic_library = crate_info.output,
+        )
+
+    link_input = cc_common.create_linker_input(
+        owner = ctx.label,
+        libraries = depset([library_to_link]),
+        user_link_flags = depset(toolchain.stdlib_linkflags),
+    )
+
+    linking_context = cc_common.create_linking_context(
+        # TODO - What to do for no_std?
```

Comment:
What do you think no_std has especially?

I only know no_std for embedded code, this also means the C++ toolchain will be different and the libs list will also match, won't it?

Reply:
Its mostly that `std` is what brings in these libs, its possible I think for someone to be using `no_std` on for instance linux for something like a kernel module or maybe an NSS hook as well as some things like SGX. I suspect that its not always going to be a toolchain thing it might be a context of compilation instead. I would see it as someone compiling C code with `-nostdlib`.

With that said, its up to how Bazel wants to view such configurations, if its expected that someone makes up a toolchain config for say `no_std` linux x86_64 and its given unique identification against toolchain matching then I fully agree that this todo can be removed.

What do you think? Is the philosophy of Bazel to assume that it is a unique toolchain or is it to allow flag flipping on the current toolchain configuration?

Reply:
I would see the no_std as a different toolchain, since it targets a different architecture (kernel mode vs userland is already a different platform in my view), but I am up for both case. I don't think Bazel has any recommendation on that (although it has the concept of platform that include more than the CPU).

For my own curiosity: Is there any actual kernel code in Rust?

Reply:
https://ldpreload.com/p/kernel-modules-in-rust-lssna2019.pdf ;)